.gitignore

```
@@ -23,11 +23,22 @@
 hs_err_pid*
 /bin/
 
+kls_database.db
+.DS_Store
+
 # Build Folders
 /build/
 /dist/
 
+# Gradle cache
+.gradle
+# /gradle/
+
 # code populated files
 /database/
 /benchmarks/
 /suitablePrgms/
+
+# IDE specific folders
+# IntelliJ
+.idea/
```

Do we want to preemptively add other IDEs & editors?

If so, it may be wise to add
```suggestion
.idea/
.vscode/
.dir-locals.el
```

---

I will implement this. Any others come to mind?

---

The .idea file was new as I was using intelliJ for it's gradle kotlin LSP. If there are other editors that people are using we can also add them as we go.

---

I would imagine there are some for Eclipse as well, but I don't use it.  I believe @charles-moloney might, however.